### PR TITLE
Added messy code for flashing the whole landed_blocks when a line is clear

### PR DESCRIPTION
**Pull request by Arrinao**
but it doesn't work

### CODE REVIEW

File: tetris.py [hunk 1]
```diff
@@ -300,13 +301,45 @@ def full_line_clear(self):
         for x_line in range(game_height):
             count = coordinates_counter[x_line]
             if count == game_width:
+                print(coordinates_counter)
+                print(y_coordinates)
+                self.flasher()
```
Comment (Akuli) [inline tetris.py:306]: When this code runs, you have already figured out which line needs to be flashed, and there's no need to do it again:

```suggestion
                self.flasher(x_line)
```

File: tetris.py [hunk 2]
```diff
@@ -300,13 +301,45 @@ def full_line_clear(self):
...
+                self.canvas.after(600)
```
Comment (Akuli) [inline tetris.py:307]: Don't use `.after()` with one argument. This is same as `time.sleep(0.6)`.

File: tetris.py [hunk 3]
```diff
@@ -300,13 +301,45 @@ def full_line_clear(self):
...
+                    for x, y in self.get_block_shape():
+                        self.canvas.create_rectangle(
+                            x * square_size,
+                            y * square_size,
+                            x * square_size + square_size,
+                            y * square_size + square_size,
+                            tags="block",
+                            fill='White',
+                        )
+                    self.canvas.after(100)
```
Comment (Akuli) [inline tetris.py:332]: Same here, this is `time.sleep(0.1)`

**Review by Akuli — CHANGES_REQUESTED** [review]

File: tetris.py [hunk 4]
```diff
@@ -300,13 +302,31 @@ def full_line_clear(self):
...
             count = coordinates_counter[x_line]
             if count == game_width:
+                for flash in range(2):
+                    self.flasher(x_line, 'yellow')
+                    self.canvas.update()
+                    time.sleep(0.1)
+                    self.flasher(x_line, 'brown')
+                    self.canvas.update()
+                    time.sleep(0.1)
                 # TODO: root.after() here
```
Comment (Akuli) [inline tetris.py:312]:
```suggestion
```

File: tetris.py [hunk 5]
```diff
@@ -300,13 +302,31 @@ def full_line_clear(self):
...
                 for letter, coord_list in self.landed_blocks.items():
                     # self.landed_blocks = {letter: [(a, b) for (a, b) in coord_list if b > x_line] + [(a, b+1) for (a, b) in coord_list if b < x_line]} #Why this doesn't work?
                     self.landed_blocks[letter] = [(a, b) for (a, b) in coord_list if b > x_line] + [
                         (a, b + 1) for (a, b) in coord_list if b < x_line
                     ]
 
+    def flasher(self, x_line, fill):
+        print('flash!')
+        for x in range(10):
```
Comment (Akuli) [inline tetris.py:321]:
```suggestion
        for x in range(game_width):
```

File: tetris.py [hunk 6]
```diff
@@ -300,13 +302,31 @@ def full_line_clear(self):
...
+                    time.sleep(0.1)
                 # TODO: root.after() here
                 for letter, coord_list in self.landed_blocks.items():
                     # self.landed_blocks = {letter: [(a, b) for (a, b) in coord_list if b > x_line] + [(a, b+1) for (a, b) in coord_list if b < x_line]} #Why this doesn't work?
                     self.landed_blocks[letter] = [(a, b) for (a, b) in coord_list if b > x_line] + [
                         (a, b + 1) for (a, b) in coord_list if b < x_line
                     ]
 
+    def flasher(self, x_line, fill):
+        print('flash!')
```
Comment (Akuli) [inline tetris.py:320]:
```suggestion
```

**Review by Akuli — CHANGES_REQUESTED** [review]

File: tetris.py [hunk 7]
```diff
@@ -21,7 +21,7 @@
 TEAL = "paleturquoise3"
 WHITE = 'snow'
 
-block_letters = ["I", "L", "L_rev", "O", "E", "Z", "Z_rev"]
+block_letters = ["I"]
```
Comment (Akuli) [inline tetris.py:24]: Did you `git diff --cached` before commit? :)

File: tetris.py [hunk 8]
```diff
@@ -297,35 +297,42 @@ def full_line_clear(self):
...
-                    self.flasher(x_line, 'brown')
-                    self.canvas.update()
-                    time.sleep(0.1)
-                self.flasher(x_line, D_GREY)
-                for letter, coord_list in self.landed_blocks.items():
-                    # self.landed_blocks = {letter: [(a, b) for (a, b) in coord_list if b > x_line] + [(a, b+1) for (a, b) in coord_list if b < x_line]} #Why this doesn't work?
-                    self.landed_blocks[letter] = [(a, b) for (a, b) in coord_list if b > x_line] + [
-                        (a, b + 1) for (a, b) in coord_list if b < x_line
+                flash_list.append(x_line)
+        if x_line in flash_list:
```
Comment (Akuli) [inline tetris.py:307]: This check is unnecessary and doesn't do what you think it does. Delete it.

File: tetris.py [hunk 9]
```diff
@@ -297,35 +297,42 @@ def full_line_clear(self):
         """
         Clears the line once it's fully populated with blocks
         """
+        flash_list = []
```
Comment (Akuli) [inline tetris.py:300]: Maybe name this `full_lines`? It's not only for flashing, it's also used for deleting lines.

File: tetris.py [hunk 10]
```diff
@@ -297,35 +297,42 @@ def full_line_clear(self):
...
+                self.canvas.update()
+                time.sleep(0.1)
+            self.canvas.delete('flash')
+        for x_line in flash_list:
+            for letter, coord_list in self.landed_blocks.items():
+                # self.landed_blocks = {letter: [(a, b) for (a, b) in coord_list if b > x_line] + [(a, b+1) for (a, b) in coord_list if b < x_line]} #Why this doesn't work?
+                self.landed_blocks[letter] = [(a, b) for (a, b) in coord_list if b > x_line] + [
+                    (a, b + 1) for (a, b) in coord_list if b < x_line
                     ]
+        print(flash_list)
```
Comment (Akuli) [inline tetris.py:322]: Delete this

File: tetris.py [hunk 11]
```diff
@@ -297,35 +297,42 @@ def full_line_clear(self):
...
-                x_line * square_size + square_size,
-                fill=fill,
-                outline=self.outline_color
-            )
+            for x_line in flash_list:
+                self.canvas.create_rectangle(
+                    x * square_size,
+                    x_line * square_size,
+                    x * square_size + square_size,
+                    x_line * square_size + square_size,
```
Comment (Akuli) [inline tetris.py:331]: These calculations are copy/pasted to many places. Can we make a new method that does them?

File: tetris.py (hunk 7 above)
Comment (Arrinao) [reply tetris.py:24]: No, but this is not a big deal. Will do it next time though :)

File: tetris.py (hunk 8 above)
Comment (Arrinao) [reply tetris.py:307]: Changed to if flash_list:

File: tetris.py (hunk 10 above)
Comment (Arrinao) [reply tetris.py:322]: Gone

File: tetris.py (hunk 11 above)
Comment (Arrinao) [reply tetris.py:331]: Sure

File: tetris.py [hunk 12]
```diff
@@ -293,20 +295,48 @@ def block_rotator(self):
...
+        Used in conjuction with full_line clear for flashing purposes
+        '''
+        for x in range(game_width):
+            for x_line in full_lines:
+                self.canvas.create_rectangle(
+                    x * square_size,
+                    x_line * square_size,
+                    x * square_size + square_size,
+                    x_line * square_size + square_size,
+                    tags="flash",
```
Comment (Akuli) [inline tetris.py:335]: The rest of the program doesn't use `flash` tag anywhere, but there is `self.canvas.delete("block")`:

```suggestion
                    tags="block",
```

**Review by Akuli — CHANGES_REQUESTED** [review]

File: tetris.py [hunk 13]
```diff
@@ -19,6 +19,7 @@
 ORANGE = "Orangered2"
 PINK = "#FF00FF"
 TEAL = "paleturquoise3"
+WHITE = "snow"
```
Comment (Akuli) [inline tetris.py:22]: delete this

**Review by Akuli — CHANGES_REQUESTED** [review]

**Pull request merged**